server/server.go

```diff
@@ -1202,9 +1202,13 @@ KELoop:
 SRLoop:
 	for i, c := range s.clients {
 		// Recover shared secrets
-		b[i].srKP, b[i].dcKP = dcnet.SharedKeys(b[i].kx, ecdh, s.sid, s.msize,
+		b[i].srKP, b[i].dcKP, err = dcnet.SharedKeys(b[i].kx, ecdh, s.sid, s.msize,
 			s.run, starts[i], c.pr.MessageCount)
-
+		if err != nil {
+			log.Printf("blaming %v for bad SharedKeys: %v", c.raddr(), err)
+			blamed = append(blamed, i)
+			continue SRLoop
```

Comment:
if this fails, we must abort blame assignment.  It doesn't make any sense to blame a particular peer at this point (after x25519 scalars have already been validated).